site/content/extend/plugins/webapp/hello-world.md

```diff
@@ -1,66 +1,157 @@
 
-Once you have npm installed, install the Mattermost Developer Kit (MDK) tool which will help you set up your workspace.
+You'll also need a Mattermost server to install and test the plugin. This server must have [Enable](https://docs.mattermost.com/administration/config-settings.html#enable-plugins) set to true in the [PluginSettings](https://docs.mattermost.com/administration/config-settings.html#plugins-beta) section of its config file. If you want to upload plugins via the System Console or API, you'll also need to set [EnableUploads](https://docs.mattermost.com/administration/config-settings.html#enable-plugin-uploads) to true in the same section.
 
-```
-npm install -g mdk
+## Setting up the Workspace
+
+Create a directory to act as your plugin workspace. With that directory, create and switch to a `webapp` directory:
+
+```bash
+mkdir webapp
+cd webapp
 ```
 
-You'll also need a Mattermost server to install and test the plugin. This server must have "Enable" and "EnableUploads" set to true in the "PluginSettings" section of its config file.
+Install the necessary NPM dependencies:
 
-## Setting up the Workspace
+```bash
+npm install --save-dev webpack webpack-cli babel-loader babel-core babel-preset-env babel-preset-react
+npm install --save react
+```
 
-To quickly set up a workspace, change into a directory where you want to store your plugins in development and run:
+Configure Babel by creating a `.babelrc` file:
 
+```json
+{
+  "presets": ["env", "react"]
+}
 ```
-mdk init plugin --name helloworld --components ProfilePopover --skip-prompts
+
+Configure Webpack by creating a `webpack.config.js` file:
+
+```js
+var path = require('path');
+var webpack = require('webpack');
+
+module.exports = {
+    entry: [
+        './src/index.jsx',
+    ],
+    resolve: {
+        extensions: ['*', '.js', '.jsx']
+    },
+    module: {
+        rules: [
+            {
+                test: /\.(js|jsx)$/,
+                exclude: /node_modules/,
+                use: 'babel-loader',
+            },
+        ],
+    },
+    externals: {
+        react: 'react',
+    },
+    output: {
+        path: path.join(__dirname, '/dist'),
+        publicPath: '/',
+        filename: 'main.js'
+    },
+};
 ```
 
-This will create your plugin workspace in a directory called `helloworld`.
+Observe that `react` is specified as an external library. This allows you to test your code locally (e.g. with [jest](https://jestjs.io/) and snapshots) but leverage the version of React shipped with Mattermost to avoid bloating your plugin.
 
-In the future, you can run `mdk init plugin` to have prompts walk you through the setup.
+Now create the entry point file and output directory:
+```bash
+mkdir src dist
+touch src/index.js
+```
 
-## Building the Plugin
+Then populate `src/index.jsx` with the following:
+```js
```

Comment:
I'd like to externalize this example into the webapp, but cutting things short for now. The actual example is also up for debate!